client/me/security-2fa-backup-codes/index.jsx

```diff
@@ -24,29 +25,48 @@ class Security2faBackupCodes extends Component {
 			showPrompt: ! printed,
 			backupCodes: [],
 			generatingCodes: false,
+			addingPassword: false,
+			lastError: null,
 		};
 	}
 
 	handleGenerateButtonClick = () => {
 		this.props.recordGoogleEvent( 'Me', 'Clicked on Generate New Backup Codes Button' );
 
+		this.setState( { addingPassword: true, showPrompt: false } );
+	};
+
+	handleGenerateButton = ( userPassword ) => {
 		this.setState( {
 			generatingCodes: true,
 			verified: false,
-			showPrompt: true,
 		} );
 
-		twoStepAuthorization.backupCodes( this.onRequestComplete );
+		twoStepAuthorization.backupCodes( userPassword, this.onRequestComplete );
+	};
+
+	toggleBackupCodePassword = ( event ) => {
+		event.preventDefault();
+		this.setState( {
+			addingPassword: ! this.state.addingPassword,
+			generatingCodes: false,
+			lastError: null,
+			showPrompt: true,
+		} );
```

Comment:
This can potentially be using an inconsistent state if multiple updates are batched together (you can read [this](https://reactjs.org/docs/react-component.html#setstate) and particularly [this](https://stackoverflow.com/questions/48563650/does-react-keep-the-order-for-state-updates/48610973#48610973) to understand why). 

To get around this, we use the callback form of `setState()`, and that way we ensure that we're using the up-to-date previous state:

```suggestion
		this.setState( ( prevState ) => ( {
			addingPassword: ! prevState.addingPassword,
			generatingCodes: false,
			lastError: null,
			showPrompt: true,
		} ) );
```